rfc/rfc021-vp_token-grant-type.md

```diff
@@ -0,0 +1,215 @@
+2. The Authorization Server MUST validate the VP and Presentation Submission according to section 6.1 of the Presentation Exchange [PE] specification.
+3. The Presentation Submission MUST be an answer to the Presentation Definition that corresponds with the requested scope.
+   How an Authorization Server determines the Presentation Definition based on scope is out of scope of this specification.
+4. A clock skew of no more than 5 seconds MAY be applied when validating the Verifiable Presentation.
+
+### 4.2 JWT format requirements
+
+1. The assertion MUST be a valid JWT according to §6.3.1 of the Verifiable Credentials Data Model 1.1 [VC-DATA-MODEL].
+2. The `vp` field of the JWT MUST be valid according to §4.10 the Verifiable Credentials Data Model 1.1 [VC-DATA-MODEL].
+3. The `iss` field MUST be a Decentralized Identifier [DID] and match the `sub` field.
+4. The `kid` header MUST be a DID URL and MUST resolve to a verificationMethod in the DID Document. The DID part of the DID URL MUST match the `iss` field.
+5. The `sub` field MUST match the `credentialSubject.id` field from all the Verifiable Credentials that are used to request the access token.
+6. The `aud` field MUST match the DID of the Authorization Server.
+7. The `iat` field MUST be present and contain a valid unix timestamp. It MUST be before the current time.
+8. The `exp` field MUST be present and contain a valid unix timestamp. It MUST be after the current time.
+9. The difference between the `exp` and `iat` fields MUST be equal or less than 5 seconds.
+10. The `jti` field MUST be present and contain a string that is unique for each access token request.
+
+### 4.3 JSON format requirements
+
+1. The assertion MUST be a valid JSON object according to §4.10 the Verifiable Credentials Data Model 1.1 [VC-DATA-MODEL].
+2. The proof of the VP MUST be a valid [JSONWebSignature2020] object. Additional requirements for the proof object:
+   1. The `challenge` field MUST be a string that is unique for each token request.
+   2. The `domain` field MUST be a DID under control of the Authorization Server.
+   3. The `verificationMethod` field MUST be a DID URL.
+   4. The id part of the DID in the `verificationMethod` field MUST match the `credentialSubject.id` field from all the Verifiable Credentials that are used to request the access token.
+   5. The `created` field MUST be present and contain a valid ISO8601 formatted date string. It MUST be before the current time.
+   6. The `expires` field MUST be present and contain a valid ISO8601 formatted date string. It MUST be after the current time.
+   7. The difference between the `expires` and `created` fields MUST be equal or less than 5 seconds.
+
+### 4.4 Preventing Token Replay
+
+The Authorization Server MUST reject any request that uses the unique value (`jti` or `challenge`) that has been used before.
```

Comment:
```suggestion
The Authorization Server MUST reject any request that uses a value for `jti` (JWT proofs) or `challenge` (JSON Web Proofs) that has been used before, by the given issuer.
```

Reply:
If it uses a value that has been used by another issuer, that would also be extremely weird.